src/topology/extremally_disconnected.lean

```diff
@@ -0,0 +1,125 @@
+/-
+Copyright (c) 2021 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+import topology.compact_open
+import topology.stone_cech
+
+/-!
+
+# Extremally disconnected spaces
+
+An extremally disconnected topological space is a space
+in which the closure of every open set is open.
+Such spaces are also called Stonean spaces.
+They are the projective objects in the category of compact Hausdorff spaces;
+this fact is proven in `TODO`.
+
+## References
+
+Gleason, Andrew M. (1958), "Projective topological spaces",
+Illinois Journal of Mathematics, 2 (4A): 482–489,
+doi:10.1215/ijm/1255454110, MR 0121775
+-/
+
+universe variables u v w
+
+noncomputable theory
+open_locale classical
+
+variables (X : Type u) [topological_space X]
+
+open function
+
+/-- An extremally disconnected topological space is a space
+in which the closure of every open set is open.
+Such spaces are also called Stonean spaces.
+They are the projective objects in the category of compact Hausdorff spaces;
+this fact is proven in `TODO`. -/
+class extremally_disconnected : Prop :=
+(open_closure : ∀ U : set X, is_open U → is_open (closure U))
+
+section
+
+include X
+/--  The assertion `compact_t2.projective` states that given continuous maps
+`f : X → Z` and `g : Y → Z` with `g` surjective between `t_2`, compact topological spaces,
+there exists a continuous lift `h : X → Y`, such that `f = g ∘ h`. -/
+def compact_t2.projective : Prop :=
+  Π {Y Z : Type u} [topological_space Y] [topological_space Z],
+  by exactI Π [compact_space Y] [t2_space Y] [compact_space Z] [t2_space Z],
+  by exactI Π {f : X → Z} {g : Y → Z} (hf : continuous f) (hg : continuous g)
```

Comment:
Is this second `by exactI` needed? I don't see any typeclasses here.

Reply:
Good catch!  I removed the second `exactI`!